enhancements/security/runtime-class-scc.md

```diff
@@ -0,0 +1,248 @@
+
+### Goals
+
+- SCC should support specifying the runtime classes that a user is allowed access to in their pod specs.
+- SCC should support specifying the runtime classes that must be specified in the pod request from a user.
+
+### Non-Goals
+
+- Changing the current SCC profiles to allow any extra runtime classes.
+- Changing behavior of the default runtime class.
+- Adding additional SCCs for various customzied behavior using AllowedRuntimeClasses.
+
+## Proposal
+
+
+[RuntimeClasses](https://kubernetes.io/docs/concepts/containers/runtime-class/) is a feature in Kubernetes that allows a user to request a certain runtime configuration. CRI-O has enhanced this feature to not only branch on different runtimes, but customize the way the pod is run. With the customization of the way a pod is run comes risk of users allocating resources they should not be allowed to. This leads to the need for a way to gate different users from using different runtime classes, and SCC is a perfect component to do so.
+
+The proposal is to add two different fields to the SCC: `AllowRuntimeClasses` and `EnsureRuntimeClasses`.
+
+### AllowRuntimeClasses
+The API for `AllowRuntimeClasses` will follow the implementation of [AllowCapabilities](https://github.com/openshift/api/blob/0ba2c3658da6cade27a5066575cd8446eb03914c/security/v1/types.go#L57..L61), where a specific SCC profile will be able to specify runtime classes that are allowed for a Pod. If a request for a runtime class comes in that doesn't match the allowed runtime classes, the Pod will be denied.
+
+However, instead of a customized type, or using a type from the kubernetes API like [the Runtime Class structure](https://github.com/kubernetes/api/blob/03aa42fe49ac2fa37646cc09e9a84fd825b88117/node/v1beta1/types.go#L37), `AllowedRuntimeClasses` will be a slice of strings. One can think of it as a list of [Handlers](https://github.com/kubernetes/api/blob/03aa42fe49ac2fa37646cc09e9a84fd825b88117/node/v1beta1/types.go#L53) of the runtime classes that are allowed.
+
+```
+type SecurityContextConstraints struct {
+    ...
+	// AllowedRuntimeClasses is a list of names of runtime classes (also called runtime handlers) that can be requested for a pod to run under.
+	// Runtime classes that match this list may be added to the pod request at the pod author's discretion.
+	// To allow all runtime classes, '*' may be specified.
+	// +nullable
+	AllowedRuntimeClasses []string `json:"allowedRuntimeClasses" protobuf:"bytes,26,rep,name=allowedRuntimeClasses"`
+    ...
+}
+```
+Fields in this slice will be logically OR'd. A user must specify an item that only matches one field to be permitted to use that runtime class. This allows an admin to specify a set of runtime classes that a user is allowed to use.
+
+### EnsuredRuntimeClasses
```

Comment:
I think `RequiredRuntimeClasses` is more appropriate for a field that doesn't do defaulting.